### PR TITLE
Bonus for reachable outpost

### PR DESCRIPTION
**Pull request by locutus2**
Give a bonus for outpost squares which in reach of a bishop or knight.

STC:
LLR: 2.96 (-2.94,2.94) [0.00,5.00]
Total: 22725 W: 4570 L: 4339 D: 13816

LTC:
LLR: 2.96 (-2.94,2.94) [0.00,5.00]
Total: 15019 W: 2333 L: 2157 D: 10529

Bench: 8503181

### CODE REVIEW

**Comment by ajithcj** [comment]
Congrats!!

**Comment by Rocky640** [comment]
Congratulations Stefan !
Now SF will play outposts at the same level as Komodo !

**Comment by mstembera** [comment]
Congrats!

**Comment by zamar** [comment]
The idea is good and solid.

However we need to improve documentation. It took me a while to figure out how the code works. However we should document what we mean by "Possible outpost", e.g. it's an outpost square where the piece can immediately move in.

Otherwise, approved.

**Comment by locutus2** [comment]
@zamar 
Your are right, i have updated the comment to clarify this.

Perhaps we should to additionally rename PossibleOutpost to something like ReachableOutpost or PotentialOutpost.

**Comment by zamar** [comment]
Yeah renaming would make sense...

**Comment by locutus2** [comment]
@zamar 

Ok, i have renamed to ReachableOutpost (feels the most fitting for me)

**Comment by glinscott** [comment]
Very nice!!  Approved.

**Comment by VoyagerOne** [comment]
Very nice indeed!

File: src/evaluate.cpp [hunk 1]
```diff
@@ -299,11 +311,16 @@ namespace {
...
-                && relative_rank(Us, s) <= RANK_6
-                && !(pos.pieces(Them, PAWN) & pawn_attack_span(Us, s)))
+            // Bonus for outpost squares
+            bb = OutpostMask[Us] & ~ei.pi->pawn_attacks_span(Them);
+            if (bb & s)
                 score += Outpost[Pt == BISHOP][!!(ei.attackedBy[Us][PAWN] & s)];
+            else
+            {
+                bb &= b & ~pos.pieces(Us);
+                if(bb)
```
Comment (hxim) [inline src/evaluate.cpp]: formatting: missing space after if

**Pull request closed without merging**